### PR TITLE
Refactor types to base

### PR DESCRIPTION
**Pull request by abhconcordium**
## Purpose
Updated `concordium-node` to parity with `concordium-base` https://github.com/Concordium/concordium-base/pull/47

Many type definitions which are used across multiple repos have been transferred to `concordium-base` from `concordium-node`. This branch removes those definitions from node, and includes the imports/refactoring required to compile using the new version of `concordium-base`.

## Changes

Multiple type definitions moved to `concordium-base`.

The JSON serialization used in the response of the `GetRewardStatus` GRPC call has been changed.
Amounts that were previously represented as integers are now represented as strings in the JSON serialization.
This is in line with how amounts are serialized elsewhere.
This is a breaking change for the tokenomics reporter.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [ ] Manual testing of affected endpoints

### CODE REVIEW

**Comment by abhconcordium** [comment]
@td202 
Only change I made from your original types file was reverting the 
`data BakerStatus` back to one where they didn't take arguments, other than that types should be compatible with your CB-1082-protocol-update branch.

This compiles with the correct concordium-base without warnings or errors. 
Any thoughts on other types which would also make sense to bring over to base?

**Comment by td202** [comment]
> 
> 
> @td202
> Only change I made from your original types file was reverting the
> `data BakerStatus` back to one where they didn't take arguments, other than that types should be compatible with your CB-1082-protocol-update branch.
> 
> This compiles with the correct concordium-base without warnings or errors.
> Any thoughts on other types which would also make sense to bring over to base?

I am moving `BakerStatus` back out of concordium-base, because this type is not part of the GRPC interface. It is only used between `Getters` and `External`, where it is used to construct an `Int64` that's passed to Rust.

File: concordium-consensus/src/Concordium/GlobalState/Basic/BlockState/AccountReleaseSchedule.hs [hunk 1]
```diff
@@ -34,113 +35,130 @@ module Concordium.GlobalState.Basic.BlockState.AccountReleaseSchedule (
...
+                    )
+                )
+        let (_pendingReleases, _totalLockedUpBalance) =
+                Vector.ifoldl'
+                    ( \acc idx -> \case
+                        Nothing -> acc
+                        Just (rel, _) ->
+                            let f (pending, am) Release{..} =
+                                    let pending' = Map.alter (maybe (Just [idx]) (Just . (idx :))) timestamp pending
+                                        am' = am + amount
```
Comment (abizjak) [inline concordium-consensus/src/Concordium/GlobalState/Basic/BlockState/AccountReleaseSchedule.hs:144]: I'm guessing only indendation/line breaks changed, but this format style seems substantially worse than the previous one.

**Review by abizjak — APPROVED** [review]

File: concordium-consensus/src/Concordium/GlobalState/Basic/BlockState/AccountReleaseSchedule.hs (hunk 1 above)
Comment (td202) [reply concordium-consensus/src/Concordium/GlobalState/Basic/BlockState/AccountReleaseSchedule.hs:144]: I don't love everything about the fourmolu formatting style, but I don't agree that it's substantially worse. I think it actually more clearly delineates the arguments of multi-line function calls such as `replicateM` and `ifoldl'` here.

**Review by td202 — APPROVED** [review]

**Pull request merged**